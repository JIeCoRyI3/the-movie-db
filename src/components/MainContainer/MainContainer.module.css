.mainContainer {
    background: #ebf6fc;
    box-sizing: border-box;
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
}

.mainContainer::after {
    content: '';
}

@media only screen and (min-width: 901px) {
    .mainContainer::after {
        flex: 0 1 30%;
    }
}

@media only screen and (max-width: 900px) {
    .mainContainer::after {
        flex: 0 1 48%;
    }
}

@media only screen and (max-width: 600px) {
    .mainContainer::after {
        flex: 0 1 100%;
    }
}
