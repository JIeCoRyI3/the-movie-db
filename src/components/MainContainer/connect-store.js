import { loadMovies } from '../../store/actions-creators/loadMovies';
import ApiClient from '../../api/apiClient';

const api = new ApiClient();

export const loadData = () => {
    return (dispatch) => {
        api.getAllMovies().then((res) => {
            console.log('ert');
            dispatch(loadMovies(res.results));
        });
    };
};
