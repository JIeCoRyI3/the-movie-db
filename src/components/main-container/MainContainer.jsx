import React from 'react';
import PropTypes from 'prop-types';
import * as styles from './MainContainer.module.css';

function MainContainer(props) {
    return <main className={styles.mainContainer}>{props.children}</main>;
}

MainContainer.propTypes = {
    children: PropTypes.arrayOf(PropTypes.object),
};

export { MainContainer };
