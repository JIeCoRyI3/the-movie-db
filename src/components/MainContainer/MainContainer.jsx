import React from 'react';
import PropTypes from 'prop-types';
import styles from './MainContainer.module.css';
import FilmsContainer from '../FilmsContainer/';
import { connect } from 'react-redux';
import NotFound from '../NotFound';
import { loadData } from './connect-store';

class MainContainer extends React.Component {
    render() {
        return (
            <main className={styles.mainContainer}>
                {this.props.films && this.props.films.length ? (
                    <FilmsContainer films={this.props.films} />
                ) : (
                    <NotFound />
                )}
            </main>
        );
    }
}

MainContainer.propTypes = {
    films: PropTypes.arrayOf(
        PropTypes.shape({
            poster_path: PropTypes.string,
            genres_ids: PropTypes.arrayOf(PropTypes.number),
            release_date: PropTypes.string,
            title: PropTypes.string.isRequired,
        })
    ),
};

const mapStateToProps = (state) => ({
    films: state.app.moviesList,
});

const mapDispatchToProps = {
    loadData,
};

const withStore = connect(mapStateToProps, mapDispatchToProps);

export default withStore(MainContainer);
