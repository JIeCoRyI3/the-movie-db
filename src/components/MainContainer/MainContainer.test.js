import React from 'react';
import { shallow, mount } from 'enzyme';
import { MainContainer } from './MainContainer';
import { findByTestId } from '@testing-library/react';

const historyMock = {
    push() {
        return 1;
    },
};
const locationMock = {
    search: '?searchBy=genreOrTitle&input=adv&sortBy=rating&sortType=asc',
};
let loadDataByGenreOrTitleMock;

beforeEach(() => {
    loadDataByGenreOrTitleMock = jest.fn();
});

describe('<MainContainer />', () => {
    it('should renders', () => {
        const component = shallow(
            <MainContainer
                loadDataByGenreOrTitle={loadDataByGenreOrTitleMock}
                history={historyMock}
                location={locationMock}
            />
        );
        expect(component).toMatchSnapshot();
    });

    xit('shoud componentDidUpdate', () => {
        const loadDataByGenreOrTitleMock = jest.fn();
        const component = mount(
            <MainContainer
                loadDataByGenreOrTitle={loadDataByGenreOrTitleMock}
                history={historyMock}
                location={locationMock}
            />
        );
        const spy = jest.spyOn(component, 'componentDidUpdate');
        component.componentDidMount();

        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
});

xdescribe('componentDidUpdate', () => {
    it('shoud enter inside', () => {
        const component = mount(
            <MainContainer
                loadDataByGenreOrTitle={loadDataByGenreOrTitleMock}
                history={historyMock}
                location={locationMock}
            />
        );

        const instance = component.instance();
        instance.props = {
            sortByRatingUp: jest.fn(),
            films: [{ 1: 'asd' }],
            byRating: true,
        };
        console.log(instance.props.sortByRatingUp);
        const spy = jest.spyOn(instance, 'sortByRating');
        instance.componentDidUpdate();

        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
});

describe('componentDidMount', () => {
    it('shoud enter inside title', () => {
        const currentLocationMock = {
            search: '?searchBy=title&input=adv&sortBy=rating&sortType=asc',
        };
        const component = shallow(
            <MainContainer
                loadDataByTitle={jest.fn()}
                loadDataByGenreOrTitle={loadDataByGenreOrTitleMock}
                history={historyMock}
                location={currentLocationMock}
            />
        );

        const instance = component.instance();
        instance.props = {
            films: [{ 1: 'asd' }],
            byRating: true,
            location: {
                search: '?searchBy=title&input=adv&sortBy=rating&sortType=asc',
            },
            loadDataByTitle: jest.fn(),
        };
        instance.componentDidMount();

        expect(instance.props.loadDataByTitle).toHaveBeenCalled();
    });

    it('shoud enter inside genre', () => {
        const currentLocationMock = {
            search: '?searchBy=genre&input=adv&sortBy=rating&sortType=asc',
        };
        const component = shallow(
            <MainContainer
                loadDataByGenre={jest.fn()}
                loadDataByGenreOrTitle={loadDataByGenreOrTitleMock}
                history={historyMock}
                location={currentLocationMock}
            />
        );

        const instance = component.instance();
        instance.props = {
            films: [{ 1: 'asd' }],
            byRating: true,
            location: {
                search: '?searchBy=genre&input=adv&sortBy=rating&sortType=asc',
            },
            loadDataByGenre: jest.fn(),
        };
        instance.componentDidMount();

        expect(instance.props.loadDataByGenre).toHaveBeenCalled();
    });

    it('shoud enter inside GenreOrTitle', () => {
        const currentLocationMock = {
            search:
                '?searchBy=genreOrTitle&input=adv&sortBy=rating&sortType=asc',
        };
        const component = shallow(
            <MainContainer
                loadDataByGenre={jest.fn()}
                loadDataByGenreOrTitle={loadDataByGenreOrTitleMock}
                history={historyMock}
                location={currentLocationMock}
            />
        );

        const instance = component.instance();
        instance.props = {
            films: [{ 1: 'asd' }],
            byRating: true,
            location: {
                search:
                    '?searchBy=genreOrTitle&input=adv&sortBy=rating&sortType=asc',
            },
            loadDataByGenreOrTitle: jest.fn(),
        };
        instance.componentDidMount();

        expect(instance.props.loadDataByGenreOrTitle).toHaveBeenCalled();
    });

    it('shoud enter inside another branch', () => {
        const currentLocationMock = {
            search: '',
        };
        const component = shallow(
            <MainContainer
                loadDataByGenre={jest.fn()}
                loadDataByGenreOrTitle={loadDataByGenreOrTitleMock}
                history={historyMock}
                location={currentLocationMock}
            />
        );

        const instance = component.instance();
        instance.props = {
            films: [{ 1: 'asd' }],
            byRating: true,
            location: {
                search: '',
            },
            loadDataByGenreOrTitle: jest.fn(),
        };
        instance.componentDidMount();

        expect(instance.props.loadDataByGenreOrTitle).not.toHaveBeenCalled();
    });
});

describe('sortBy', () => {
    it('shoud call sortByRating asc', () => {
        const currentLocationMock = {
            search:
                '?searchBy=genreOrTitle&input=adv&sortBy=rating&sortType=asc',
        };
        const component = shallow(
            <MainContainer
                sortByRatingUp={jest.fn}
                loadDataByGenreOrTitle={loadDataByGenreOrTitleMock}
                history={historyMock}
                location={currentLocationMock}
            />
        );

        const instance = component.instance();
        instance.props = {
            films: [{ 1: 'asd' }],
            byRating: true,
            sortByRatingUp: jest.fn(),
            sortByReleaseDateUp: jest.fn(),
        };
        const spy = jest.spyOn(instance, 'sortByRating');
        instance.componentDidUpdate();

        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });

    it('shoud call sortByRating desc', () => {
        const currentLocationMock = {
            search:
                '?searchBy=genreOrTitle&input=adv&sortBy=rating&sortType=desc',
        };
        const component = shallow(
            <MainContainer
                sortByRatingDown={jest.fn}
                loadDataByGenreOrTitle={loadDataByGenreOrTitleMock}
                history={historyMock}
                location={currentLocationMock}
            />
        );

        const instance = component.instance();
        instance.props = {
            films: [{ 1: 'asd' }],
            byRating: true,
            sortByRatingDown: jest.fn(),
        };
        const spy = jest.spyOn(instance, 'sortByRating');
        instance.componentDidUpdate();

        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });

    it('shoud call sortByDate asc', () => {
        const currentLocationMock = {
            search: '?searchBy=genreOrTitle&input=adv&sortBy=date&sortType=asc',
        };
        const component = shallow(
            <MainContainer
                sortByReleaseDateUp={jest.fn()}
                loadDataByGenreOrTitle={loadDataByGenreOrTitleMock}
                history={historyMock}
                location={currentLocationMock}
            />
        );

        const instance = component.instance();
        instance.props = {
            films: [{ 1: 'asd' }],
            byRating: true,
            sortByReleaseDateUp: jest.fn(),
        };
        const spy = jest.spyOn(instance, 'sortByDate');
        instance.componentDidUpdate();

        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });

    it('shoud call sortByDate desc', () => {
        const currentLocationMock = {
            search:
                '?searchBy=genreOrTitle&input=adv&sortBy=date&sortType=desc',
        };
        const component = shallow(
            <MainContainer
                sortByReleaseDateDown={jest.fn}
                loadDataByGenreOrTitle={loadDataByGenreOrTitleMock}
                history={historyMock}
                location={currentLocationMock}
            />
        );

        const instance = component.instance();
        instance.props = {
            films: [{ 1: 'asd' }],
            byRating: true,
            sortByReleaseDateDown: jest.fn(),
        };
        const spy = jest.spyOn(instance, 'sortByDate');
        instance.componentDidUpdate();

        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
});
