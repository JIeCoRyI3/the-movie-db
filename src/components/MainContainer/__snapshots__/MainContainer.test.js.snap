// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainContainer /> should renders 1`] = `
<main>
  <NotFound />
</main>
`;
